pkg/operator/configobservation/configobservercontroller/observe_config_controller.go

```diff
@@ -34,6 +37,7 @@ func NewConfigObserver(
 				InfrastructureLister: configinformers.Config().V1().Infrastructures().Lister(),
 				NetworkLister:        configinformers.Config().V1().Networks().Lister(),
 				ResourceSync:         resourceSyncer,
+				ConfigMapLister:      configMapLister,
```

Comment:
looks like you forgot to wait for a sync and you forgot to wire up to the informer to add an event handler.